evalml/pipelines/component_graph.py

```diff
@@ -455,8 +465,20 @@ def _transform_features(
                 else:
                     output = component_instance.predict(x_inputs)
                 output_cache[f"{component_name}.x"] = output
+            if self.cached_data is not None and fit:
+                self.component_instances[component_name] = component_instance
+
         return output_cache
 
+    def _get_stacked_ensemble_component(self, hashes, component_name, fit):
```

Comment:
nit: might be misunderstanding here but I think we should rename this method if it's also used to get components other than the stacked ensemble component